### DISCUSSION TITLE
How to calculate capacitance for setRC.tcl

### DISCUSSION THREAD

**Discussion by lokki11199** [The-OpenROAD-Project/OpenROAD#3429] [Q&A] [answered]
Hello! I need to calculate per-lenght capacitance for "set_layer_rc" in setRC.tcl. It doesn't present at PDK and I am trying to calculate it from the data presented at tech.lef file. Can someone describe me how can I do it?

**Comment by vijayank88** [comment]
Refer sky130hd [setRC.tcl](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/platforms/sky130hd/setRC.tcl).
Refer the document [here](https://github.com/The-OpenROAD-Project/OpenROAD/blob/2f3991101c2414dac9040684a822f16fe19e825f/src/rsz/README.md?plain=1#L32)

**Reply by lokki11199** [reply]
Could you give me a formula, beacuse I am using (CAPACITANCE CPERSQDIST x segment width x segment length) +
(EDGECAPACITANCE x 2 (segment width + segment length)), where segment length is 1 and I get an incorrect result.

**Reply by rovinski** [reply, marked as answer]
If the parasitics info is already in the tech LEF file, then you don't need to add a setRC.tcl on top of it unless you are trying to override the values from the tech LEF.

That being said, we have found that only very old processes store their parasitics in the tech LEF and the numbers presented don't tend to correlate well with the post-extraction netlist. It is fine for prototyping, but you might want to look into ORFS [correlateRC.py](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/bb925a3dfcf5c3c11d8c04bdcc9c73416ab8c12b/flow/util/correlateRC.py) to for better parasitics correlation.

**Reply by lokki11199** [reply]
Then why does Nangate45's setRC.tcl have the "set_layer_rc" parameter, although CAPACITANCE CPERSQDIST, EDGECAPACITANCE and RESISTANCE RPERSQ are explicitly defined in Nangate45's tech LEF?

**Reply by maliberty** [reply]
There is some assumed density built into those values. We found they didn't correlate as well to OR's results
